### PR TITLE
[DEV-1140] get kyc file from customer

### PR DESCRIPTION
**Pull request by Kolibri1990**
#### Pre-Release

- [ ] Check migrations
  - No database related infos (sqldb-xxx)
  - Impact on GS (new/removed columns)
- [ ] Check for linter errors (in PR)
- [ ] Test basic user operations
  - Login/logout
  - Registration
  - Update user data
  - Create/delete payment routes

#### Post-Release

- Test basic user operations
- Monitor application insights log

### CODE REVIEW

File: src/subdomains/generic/user/models/kyc/kyc.service.ts [hunk 1]
```diff
@@ -287,4 +291,65 @@ export class KycService {
...
+    return {
+      documents: customerInfos
+        .filter(
+          (document) =>
+            ([KycDocument.ONLINE_IDENTIFICATION, KycDocument.CHATBOT_ONBOARDING].includes(document.document) &&
+              document.label == 'Report') ||
+            document.document == KycDocument.INCORPORATION_CERTIFICATE,
+        )
+        .map((d) => d.document),
+    };
```
[the remark this answers is not in the document]
Comment (davidleomay) [reply src/subdomains/generic/user/models/kyc/kyc.service.ts:316]: Maybe create a mapping method with the new DocumentType?
```ts
private getDocuments(type: KycDocumentType, documents: DocumentInfo[]): DocumentInfo | undefined {
  switch (type) {
    case KycDocumentType.IDENTIFICATION:
      return documents.find((d) => d.document === KycDocument.ONLINE_IDENTIFICATION && d.label === 'Report');
    case ...
  }
}
```

[the remark this answers is not in the document]
Comment (davidleomay) [reply src/subdomains/generic/user/models/kyc/kyc.service.ts:316]: And we need to get the video identification document, if no online id is found.

File: src/subdomains/generic/user/models/kyc/kyc.service.ts [hunk 2]
```diff
@@ -287,4 +291,65 @@ export class KycService {
...
+  }
+
+  async getKycFile(userId: number, walletId: number, kycDocument: KycDocument): Promise<any> {
+    const wallet = await this.walletRepo.findOne({ where: { id: walletId } });
+    const user = await this.userRepo.findOne({ where: { id: userId }, relations: ['userData', 'wallet'] });
+    if (user.wallet.id != wallet.id) throw new ForbiddenException('User not from wallet');
```
Comment (davidleomay) [inline src/subdomains/generic/user/models/kyc/kyc.service.ts:322]:
```ts
const user = await this.userRepo.findOne({ where: { id: userId, wallet: { id: walletId }, relations: ['userData', 'wallet'] });
if (!user) throw new NotFoundException('User not found');
```

File: src/subdomains/generic/user/models/kyc/kyc.controller.ts [hunk 3]
```diff
@@ -38,6 +48,34 @@ export class KycController {
     await this.kycService.triggerWebhook(dto.userDataId, dto.reason);
   }
 
+  // --- KYC COMPANY Calls //
+  @Get('kycData')
```
[the remark this answers is not in the document]
Comment (Kolibri1990) [reply src/subdomains/generic/user/models/kyc/kyc.controller.ts:52]: Good idea... just moved it from wallet to kyc controller

**Review by davidleomay — APPROVED** [review]

**Pull request merged**